storage-image-classification/functions/src/index.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as functions from "firebase-functions";
+import vision from "@google-cloud/vision";
+
+import config from "./config";
+
+const admin = require("firebase-admin");
+admin.initializeApp();
+
+const client = new vision.ImageAnnotatorClient();
+const db = admin.firestore();
+
+exports.labelImage = functions.storage.object().onFinalize(async (object) => {
+  // TODO: allow configuration.
+  if (
+    !object.name?.toLowerCase().endsWith(".jpg") &&
+    !object.name?.toLowerCase().endsWith(".jpeg") &&
+    !object.name?.toLowerCase().endsWith(".png")
```

Comment:
consider checking contentType instead of specific file extension, like storage-resize-images:

https://github.com/firebase/extensions/blob/596a9f2b07ed0eb81fb6c0256cb6b61a1b6ad8e4/storage-resize-images/functions/src/index.ts#L57

Reply:
Done!